examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -142,20 +142,31 @@ class _PestoDemoState extends State<PestoDemo> {
           new DrawerItem(
             child: new Text('Home'),
             selected: !config.showFavorites,
-            onPressed: () => Navigator.pushNamed(context, PestoDemo.routeName)
+            onPressed: () {
+              Navigator.popUntil(context, (Route<dynamic> route) {
```

Comment:
Yes, that's effectively what happens.  With `Navigator.popUntil(context, ModalRoute.matches('/pesto'))` there's no need for case analysis and the whole thing is pretty tidy. As apps get more complex (more routes) I think this style of targeted popping might be a good pattern to encourage.